buildSrc/src/main/groovy/io/spring/gradle/convention/JavadocApiPlugin.groovy

```diff
@@ -49,6 +46,13 @@ public class JavadocApiPlugin implements Plugin<Project> {
 
 		api.setGroup("Documentation");
 		api.setDescription("Generates aggregated Javadoc API documentation.");
+		api.doLast {
+			if (JavaVersion.current().isJava11Compatible()) {
+				File packageList = new File(api.destinationDir, 'package-list')
```

Comment:
It is better to use project.file to create references to files as it is more robust. See https://docs.gradle.org/current/userguide/working_with_files.html#sec:single_file_paths

> Never use new File(relative path) because this creates a path relative to the current working directory (CWD). Gradle can make no guarantees about the location of the CWD, which means builds that rely on it may break at any time.

Obviously this works if the path is absolute, but absolute paths have the problem of causing a cache miss if the build happens in one directory and then tries to run again in another directory. This is a common scenario when using Gradle enterprise cache (which we use) with a build happening on one machine that adds a cache entry and then someone building on another machine in a different directory.

Reply:
I think it would be preferable to use [Project.copy](https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html#copy-org.gradle.api.Action-) to copy the files as it is build in.